src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -467,63 +466,15 @@ public void TestWorkingDirectoryPropertyInChildProcess()
         [OuterLoop("Requires admin privileges")]
         public void TestUserCredentialsPropertiesOnWindows()
         {
-            const string username = "testForDotnetRuntime";
-            using WindowsTestAccount testAccount = new WindowsTestAccount(username);
-
-            bool hasStarted = false;
-            SafeProcessHandle handle = null;
-            Process p = null;
-            string workingDirectory = null;
-
-            try
-            {
-                p = CreateProcessLong();
-
-                workingDirectory = string.IsNullOrEmpty(p.StartInfo.WorkingDirectory)
-                    ? Directory.GetCurrentDirectory()
-                    : p.StartInfo.WorkingDirectory;
-
-                if (PlatformDetection.IsNotWindowsServerCore) // for this particular Windows version it fails with Attempted to perform an unauthorized operation (#46619)
-                {
-                    // ensure the new user can access the .exe (otherwise you get Access is denied exception)
-                    SetAccessControl(username, p.StartInfo.FileName, workingDirectory, add: true);
-                }
-
-                p.StartInfo.LoadUserProfile = true;
-                p.StartInfo.UserName = username;
-                p.StartInfo.PasswordInClearText = testAccount.Password;
-
-                try
-                {
-                    hasStarted = p.Start();
-                }
-                catch (Win32Exception ex) when (ex.NativeErrorCode == ERROR_SHARING_VIOLATION)
-                {
-                    throw new SkipTestException($"{p.StartInfo.FileName} has been locked by some other process");
-                }
+            using Process p = CreateProcessLong();
+            p.StartInfo.LoadUserProfile = true;
 
+            using var testAccountCleanup = CreateUserAndExecute(p);
 
-                Assert.Equal(username, Helpers.GetProcessUserName(p));
-                bool isProfileLoaded = GetNamesOfUserProfiles().Any(profile => profile.Equals(username));
-                Assert.True(isProfileLoaded);
-            }
-            finally
-            {
-                if (handle != null)
-                    handle.Dispose();
-
-                if (hasStarted)
-                {
-                    p.Kill();
-
-                    Assert.True(p.WaitForExit(WaitInMS));
-                }
-
-                if (PlatformDetection.IsNotWindowsServerCore)
-                {
-                    SetAccessControl(username, p.StartInfo.FileName, workingDirectory, add: false); // remove the access
-                }
-            }
+            string username = testAccountCleanup.ProcessAccountName.Split('\\').Last();
+            Assert.Equal(username, Helpers.GetProcessUserName(p));
+            bool isProfileLoaded = GetNamesOfUserProfiles().Any(profile => profile.Equals(username));
+            Assert.True(isProfileLoaded);
```

Comment:
this refactor has made it much simpler to read and understand the code, great! 👍